substrate/frame/revive/src/exec.rs

```diff
@@ -1859,7 +1866,7 @@ mod tests {
 		) -> H256 {
 			Loader::mutate(|loader| {
 				// Generate code hashes as monotonically increasing values.
-				let hash = <Test as frame_system::Config>::Hash::from_low_u64_be(loader.counter);
+				let hash = <Test as frame_system::Config>::Hash::from_low_u64_be(loader.counter + 1);
```

Comment:
In the context of this PR it is obvious as to why this is needed. But for an outsider I don't think it's obvious too. Can you please extend the comment above?